src/traits.rs

```diff
@@ -59,3 +59,18 @@ impl<'a,T> Strided for MutStride<'a,T> {
 impl<'a,T> MutStrided for MutStride<'a,T> {
     fn as_stride_mut(&mut self) -> MutStride<T> { self.reborrow() }
 }
+
+// this isn't as general as it could be.
+impl<'a,T, X: Deref<Target=[T]>> From<&'a X> for Stride<'a,T> {
```

Comment:
I think this would be slightly more general as

```rust
impl<'a, T, X: AsRef<[T]>> From<&'a X> for Stride<'a, T> {

}
```

And similarly with `AsMut` for the mutability one.